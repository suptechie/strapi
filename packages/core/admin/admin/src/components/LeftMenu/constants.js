export const SET_SECTION_LINKS = 'StrapiAdmin/LeftMenu/SET_SECTION_LINKS';
export const UNSET_IS_LOADING = 'StrapiAdmin/LeftMenu/UNSET_IS_LOADING';
