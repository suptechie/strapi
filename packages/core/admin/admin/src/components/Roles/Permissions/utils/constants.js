export const cellWidth = '120px';
export const firstRowWidth = '200px';
