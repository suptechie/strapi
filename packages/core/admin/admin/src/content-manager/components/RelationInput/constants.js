export const RELATION_ITEM_HEIGHT = 58;
