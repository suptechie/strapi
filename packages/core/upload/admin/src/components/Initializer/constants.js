// eslint-disable-next-line import/prefer-default-export
export const SET_FILE_MODE_TIMESTAMPS = 'UPLOAD/INITIALIZER/SET_FILE_MODE_TIMESTAMPS';
