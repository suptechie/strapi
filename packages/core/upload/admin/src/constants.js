export const AssetType = {
  Video: 'video',
  Image: 'image',
  Document: 'doc',
  Unknown: 'unknown', // useful when CORS prevents from getting the mime type of an asset
};

export const AssetSource = {
  Url: 'url',
  Computer: 'computer',
};
