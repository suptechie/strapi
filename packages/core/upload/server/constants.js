'use strict';

const ACTIONS = {
  read: 'plugin::upload.read',
  readSettings: 'plugin::upload.settings.read',
  create: 'plugin::upload.assets.create',
  update: 'plugin::upload.assets.update',
  download: 'plugin::upload.assets.download',
  copyLink: 'plugin::upload.assets.copy-link',
};

module.exports = {
  ACTIONS,
};
