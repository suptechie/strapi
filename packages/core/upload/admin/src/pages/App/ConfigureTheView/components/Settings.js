import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { Grid, GridItem } from '@strapi/design-system/Grid';
import { Select, Option } from '@strapi/design-system/Select';
import getTrad from '../../../../utils/getTrad';
import { pageSizes, sortOptions } from '../../../../constants';

const Settings = ({ sort = '', pageSize = 10, onChange }) => {
  const { formatMessage } = useIntl();

  return (
    <Grid gap={4}>
      <GridItem s={12} col={6}>
        <Select
          label={formatMessage({
            id: getTrad('config.entries.title'),
            defaultMessage: 'Entries per page',
          })}
          hint={formatMessage({
            id: getTrad('config.note'),
            defaultMessage: 'Note: You can override this value in the media library.',
          })}
          onChange={(value) => onChange({ target: { name: 'pageSize', value } })}
          name="pageSize"
          value={pageSize}
          test-pageSize={pageSize}
          data-testid="pageSize-select"
        >
          {pageSizes.map((pageSize) => (
            <Option data-testid={`pageSize-option-${pageSize}`} key={pageSize} value={pageSize}>
              {pageSize}
            </Option>
          ))}
        </Select>
      </GridItem>
      <GridItem s={12} col={6}>
        <Select
          label={formatMessage({
            id: getTrad('config.sort.title'),
            defaultMessage: 'Default sort order',
          })}
          hint={formatMessage({
            id: getTrad('config.note'),
            defaultMessage: 'Note: You can override this value in the media library.',
          })}
          onChange={(value) => onChange({ target: { name: 'sort', value } })}
          name="sort"
          value={sort}
          test-sort={sort}
          data-testid="sort-select"
        >
          {sortOptions.map((filter) => (
            <Option
              data-testid={`sort-option-${filter.value}`}
              key={filter.key}
              value={filter.value}
            >
              {formatMessage({ id: getTrad(filter.key), defaultMessage: `${filter.value}` })}
            </Option>
          ))}
        </Select>
      </GridItem>
    </Grid>
  );
};

Settings.propTypes = {
  sort: PropTypes.string.isRequired,
  pageSize: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};

export { Settings };
