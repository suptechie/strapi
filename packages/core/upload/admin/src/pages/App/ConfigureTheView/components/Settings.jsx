import React from 'react';

import { Box, Grid, SingleSelectOption, SingleSelect, Field } from '@strapi/design-system';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { pageSizes, sortOptions } from '../../../../constants';
import getTrad from '../../../../utils/getTrad';

const Settings = ({ sort = '', pageSize = 10, onChange }) => {
  const { formatMessage } = useIntl();

  return (
    <Box
      background="neutral0"
      hasRadius
      shadow="tableShadow"
      paddingTop={6}
      paddingBottom={6}
      paddingLeft={7}
      paddingRight={7}
    >
      <Grid.Root gap={4}>
        <Grid.Item s={12} col={6}>
          <Field.Root
            hint={formatMessage({
              id: getTrad('config.entries.note'),
              defaultMessage: 'Number of assets displayed by default in the Media Library',
            })}
            name="pageSize"
          >
            <Field.Label>
              {formatMessage({
                id: getTrad('config.entries.title'),
                defaultMessage: 'Entries per page',
              })}
            </Field.Label>
            <SingleSelect
              onChange={(value) => onChange({ target: { name: 'pageSize', value } })}
              value={pageSize}
            >
              {pageSizes.map((pageSize) => (
                <SingleSelectOption key={pageSize} value={pageSize}>
                  {pageSize}
                </SingleSelectOption>
              ))}
            </SingleSelect>
            <Field.Hint />
          </Field.Root>
        </Grid.Item>
        <Grid.Item s={12} col={6}>
          <Field.Root
            hint={formatMessage({
              id: getTrad('config.note'),
              defaultMessage: 'Note: You can override this value in the media library.',
            })}
            name="sort"
          >
            <Field.Label>
              {formatMessage({
                id: getTrad('config.sort.title'),
                defaultMessage: 'Default sort order',
              })}
            </Field.Label>
            <SingleSelect
              onChange={(value) => onChange({ target: { name: 'sort', value } })}
              value={sort}
              test-sort={sort}
              data-testid="sort-select"
            >
              {sortOptions.map((filter) => (
                <SingleSelectOption
                  data-testid={`sort-option-${filter.value}`}
                  key={filter.key}
                  value={filter.value}
                >
                  {formatMessage({ id: getTrad(filter.key), defaultMessage: `${filter.value}` })}
                </SingleSelectOption>
              ))}
            </SingleSelect>
            <Field.Hint />
          </Field.Root>
        </Grid.Item>
      </Grid.Root>
    </Box>
  );
};

Settings.propTypes = {
  sort: PropTypes.string.isRequired,
  pageSize: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
};

export { Settings };
