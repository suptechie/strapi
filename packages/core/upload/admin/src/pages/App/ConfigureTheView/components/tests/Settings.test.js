import React from 'react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { render as renderRTL } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TrackingProvider } from '@strapi/helper-plugin';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';

import { Settings } from '../Settings';
import { pageSizes, sortOptions } from '../../../../../constants';

const testPageSize = pageSizes[0];
const testSort = sortOptions[0].value;

const render = ({ pageSize = testPageSize, sort = testSort, onChange = jest.fn() } = {}) => ({
  user: userEvent.setup(),
  ...renderRTL(<Settings sort={sort} pageSize={pageSize} onChange={onChange} />, {
    wrapper: ({ children }) => (
      <IntlProvider locale="en" messages={{}}>
        <TrackingProvider>
          <ThemeProvider theme={lightTheme}>
            <MemoryRouter>{children}</MemoryRouter>
          </ThemeProvider>
        </TrackingProvider>
      </IntlProvider>
    ),
  }),
});

describe('Upload - Configure | Settings', () => {
  beforeEach(() => {
    jest.restoreAllMocks();
  });

  describe('initial render', () => {
    it('renders and matches the snapshot', async () => {
      const { getByText, container } = render();

      expect(getByText('Entries per page')).toBeInTheDocument();

      expect(container).toMatchSnapshot();
    });
  });

  describe('pageSize', () => {
    it('renders all page Sizes', async () => {
      const { getByRole, user } = render();

      await user.click(getByRole('combobox', { name: 'Entries per page' }));

      pageSizes.forEach((size) => {
        expect(getByRole('option', { name: size })).toBeInTheDocument();
      });
    });

    it('call onChange when changing page size', async () => {
      const onChange = jest.fn();
      const { user, getByRole } = render({ onChange });

      const testValue = pageSizes[1];

      await user.click(getByRole('combobox', { name: 'Entries per page' }));

      await user.click(getByRole('option', { name: testValue }));

      expect(onChange).toHaveBeenCalledTimes(1);
      expect(onChange).toHaveBeenCalledWith({ target: { name: 'pageSize', value: testValue } });
    });
  });

  describe('sort', () => {
    it('renders all sort options', async () => {
      const { user, getByRole } = render();

      await user.click(getByRole('combobox', { name: 'Default sort order' }));

      sortOptions.forEach((filter) => {
        expect(getByRole('option', { name: filter.value })).toBeInTheDocument();
      });
    });

    it('call onChange when changing sort', async () => {
      const onChange = jest.fn();
      const { user, getByRole } = render({ onChange });

      const testValue = sortOptions[1].value;

      await user.click(getByRole('combobox', { name: 'Default sort order' }));

      await user.click(getByRole('option', { name: testValue }));

      expect(onChange).toHaveBeenCalledTimes(1);
      expect(onChange).toHaveBeenCalledWith({ target: { name: 'sort', value: testValue } });
    });
  });
});
