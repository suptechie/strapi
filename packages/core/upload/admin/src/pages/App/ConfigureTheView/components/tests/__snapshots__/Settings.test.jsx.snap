// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload - Configure | Settings initial render renders and matches the snapshot 1`] = `
.c0 {
  background: #ffffff;
  padding-block-start: 24px;
  padding-inline-end: 32px;
  padding-block-end: 24px;
  padding-inline-start: 32px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33, 33, 52, 0.1);
}

.c4 {
  display: flex;
}

.c6 {
  background: #ffffff;
  padding-block-start: 8px;
  padding-inline-end: 12px;
  padding-block-end: 8px;
  padding-inline-start: 12px;
  border-radius: 4px;
  position: relative;
  overflow: hidden;
  cursor: default;
}

.c9 {
  flex: 1;
}

.c3 {
  align-items: stretch;
  display: flex;
  flex-direction: column;
  gap: 4px;
}

.c7 {
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 16px;
  justify-content: space-between;
}

.c10 {
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 12px;
}

.c5 {
  font-size: 1.2rem;
  line-height: 1.33;
  font-weight: 600;
  color: #32324d;
}

.c11 {
  font-size: 1.4rem;
  line-height: 1.43;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #32324d;
}

.c15 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: #666687;
}

.c16 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c8 {
  border: 1px solid #dcdce4;
  outline: none;
  box-shadow: 0;
  transition-property: border-color,box-shadow,fill;
  transition-duration: 0.2s;
}

.c8[aria-disabled='true'] {
  color: #666687;
}

.c8:focus-visible {
  outline: none;
}

.c8:focus-within {
  border: 1px solid #4945ff;
  box-shadow: #4945ff 0px 0px 0px 2px;
}

.c14>svg {
  fill: #666687;
}

.c12 {
  flex: 1;
  font-size: 1.4rem;
  line-height: 2.2rem;
}

.c13 {
  display: flex;
  gap: 4px;
  flex-wrap: wrap;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  gap: 16px;
}

.c2 {
  grid-column: span 6;
  max-width: 100%;
}

@media (max-width: 1100px) {
  .c2 {
    grid-column: span 12;
  }
}

@media (max-width: 550px) {
  .c2 {
    grid-column: span;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <label
            alignitems="center"
            class="c4 c5"
            for=":r0:"
            id=":r0:-label"
          >
            Entries per page
          </label>
          <div
            aria-autocomplete="none"
            aria-controls="radix-:r2:"
            aria-describedby=":r0:-hint"
            aria-expanded="false"
            aria-labelledby=":r0:-label"
            aria-required="false"
            class="c6 c7 c8"
            data-state="closed"
            dir="ltr"
            name="pageSize"
            role="combobox"
            tabindex="0"
          >
            <span
              class="c9 c10"
            >
              <span
                class="c11 c12"
              >
                <span
                  class="c13"
                >
                  10
                </span>
              </span>
            </span>
            <span
              class="c10"
            >
              <span
                aria-hidden="true"
                class="c14"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                  />
                </svg>
              </span>
            </span>
          </div>
          <p
            class="c15"
            id=":r0:-hint"
          >
            Number of assets displayed by default in the Media Library
          </p>
        </div>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <label
            alignitems="center"
            class="c4 c5"
            for=":r3:"
            id=":r3:-label"
          >
            Default sort order
          </label>
          <div
            aria-autocomplete="none"
            aria-controls="radix-:r5:"
            aria-describedby=":r3:-hint"
            aria-expanded="false"
            aria-labelledby=":r3:-label"
            aria-required="false"
            class="c6 c7 c8"
            data-state="closed"
            dir="ltr"
            name="sort"
            role="combobox"
            tabindex="0"
          >
            <span
              class="c9 c10"
            >
              <span
                class="c11 c12"
              >
                <span
                  class="c13"
                >
                  createdAt:DESC
                </span>
              </span>
            </span>
            <span
              class="c10"
            >
              <span
                aria-hidden="true"
                class="c14"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                  />
                </svg>
              </span>
            </span>
          </div>
          <p
            class="c15"
            id=":r3:-hint"
          >
            Note: You can override this value in the media library.
          </p>
        </div>
      </div>
    </div>
  </div>
  <span
    class="c16"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
